Makefile

```diff
@@ -1,183 +1,397 @@
 
-.PHONY: ci-go
-ci-go: ## Creates, provisions, tests, and destroys GCE host for testing staging environment.
-	./devops/gce-nested/ci-go.sh
-
-.PHONY: ci-teardown
-ci-teardown: ## Destroys GCE host for testing staging environment.
-	./devops/gce-nested/gce-stop.sh
-
-.PHONY: ci-deb-tests
-ci-deb-tests: ## Runs deb tests in ci
-	@./devops/scripts/test-built-packages.sh
+.PHONY: update-pip-requirements
+update-pip-requirements: update-admin-pip-requirements update-python3-requirements ## Update all requirements with pip-compile.
 
-.PHONY: install-mypy
-install-mypy: ## pip install mypy in a dedicated python3 virtualenv
-	if [[ ! -d .python3/.venv ]] ; then \
-	  virtualenv --python=python3 .python3/.venv && \
-	  .python3/.venv/bin/pip3 install mypy ; \
-	fi
 
-.PHONY: typelint
-typelint: install-mypy ## Runs type linting
-	.python3/.venv/bin/mypy ./securedrop ./admin
-	.python3/.venv/bin/mypy --disallow-incomplete-defs --disallow-untyped-defs ./securedrop/rm.py
+#################
+#
+# Static analysis
+#
+#################
 
 .PHONY: ansible-config-lint
-ansible-config-lint: ## Runs custom Ansible env linting tasks.
-	molecule verify -s ansible-config
+ansible-config-lint: venv ## Run custom Ansible linting tasks.
+	@echo "███ Linting Ansible configuration..."
+	@. $(VENV)/bin/activate && $(VENV)/bin/molecule verify -s ansible-config
+	@echo
 
-.PHONY: docs
-docs: ## Build project documentation in live reload for editing
-# Spins up livereload environment for editing; blocks.
-	make -C docs/ clean && sphinx-autobuild docs/ docs/_build/html
+.PHONY: app-lint
+app-lint: venv ## Test pylint compliance.
+	@echo "███ Linting application code..."
+	@cd securedrop && find . -name '*.py' | xargs $(VENV)/bin/pylint --reports=no --errors-only \
+	   --disable=no-name-in-module \
+	   --disable=unexpected-keyword-arg \
+	   --disable=too-many-function-args \
+	   --disable=import-error \
+	   --disable=no-member \
+	   --max-line-length=100
+	@echo
+
+.PHONY: app-lint-full
+app-lint-full: venv ## Test pylint compliance, with no checks disabled.
+	@echo "███ Linting application code with no checks disabled..."
+	@cd securedrop && find . -name '*.py' | xargs $(VENV)/bin/pylint
+	@echo
 
 .PHONY: docs-lint
-docs-lint: ## Check documentation for common syntax errors.
+docs-lint: venv ## Check documentation for common syntax errors.
+	@echo "███ Linting documentation..."
 # The `-W` option converts warnings to errors.
 # The `-n` option enables "nit-picky" mode.
-	make -C docs/ clean && sphinx-build -Wn docs/ docs/_build/html
+	@. $(VENV)/bin/activate && make -C docs/ clean && sphinx-build -Wn docs/ docs/_build/html
+	@echo
 
 .PHONY: docs-linkcheck
-docs-linkcheck: ## Check documentation for broken links
-	make -C docs/ clean && sphinx-build -b linkcheck -Wn docs/ docs/_build/html
+docs-linkcheck:  ## Check documentation for broken links.
+	@. $(VENV)/bin/activate && make -C docs/ clean && sphinx-build -b linkcheck -Wn docs/ docs/_build/html
```

Comment:
summary of synchronous discussion regarding letting devs manage their virtualenvs:

1. add a note in the dev docs that the default virtualenv should be python 3 
2. update the paths in the dev docs to the python3/requirements files 
3. add a small note in the docs that "note: this project uses python 2.7 in production and tests in CI are also ran in python 2.7" - we could optionally add much more to this section, but the idea here is that soon after SecureDrop 0.14.0 is released, we won't need to have `master` contain any python 2.7 compliant code anymore for the webapps (to be clear: SecureDrop 0.14.0 is not shipping Python 3 in prod, but around the time of its release we will have at least a wip PR up with the migration path)
2. we'll remove the `$(VENV)/bin/activate` from the makefile and let the dev activate the Python 3 virtualenv

Reply:
Doc notes added, activation removed.